### PR TITLE
chore(core): remove @View annotation

### PR DESCRIPTION
**Pull request by btford**
- **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit-message-format
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Removes deprecated `@View` annotation API
- **What is the current behavior?** (You can also link to an open issue here)

Related to #7363
- **What is the new behavior (if this is a feature change)?**
- **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
- **Other information**:

### CODE REVIEW

File: modules/angular2/src/core/metadata.ts [hunk 1]
```diff
@@ -918,7 +914,7 @@ export var Directive: DirectiveFactory = <DirectiveFactory>makeDecorator(Directi
  * }
  * ```
  */
-export var View: ViewFactory =
```
Comment (mhevery) [inline modules/angular2/src/core/metadata.ts]: I think you can leave the export here. (Just remove it from the public surface)

Comment (mhevery) [inline modules/angular2/src/core/metadata.ts]: I see the public one is exported as `*`. So this is fine. (ignore the comment)

**Comment by mhevery** [comment]
LGTM, can you prepare the google3?

**Comment by btford** [comment]
Waiting on one travis flake and one internal flake, but otherwise this should be good to go.

@tbosch might be interested in rebasing on top of this quickly after it lands.

**Comment by PascalPrecht** [comment]
@btford @mhevery prolly already too late, but this is a breaking change as `@View` has been optional, now it's gone.

I know some ppl are relying on it. Can we update the commit message, before this goes into master?

**Pull request closed without merging**